bip-0432.mediawiki

```diff
@@ -0,0 +1,141 @@
+allows a small amount of contract value to be given to an output to allow network relay
+by passing dust checks, but not as the primary source of fee funds. Instead the child transaction
+spending the anchor is responsible for providing the funds.
+
+In this and similar abstractions it would be cleaner if the anchor itself could be 0-value
+instead of requiring anchors to exceed dust amounts.
+
+===Example Use Cases===
+
+* Batched payments with segregated fee pools: Batched payments that can be fee bumped without access to customer deposit-related key material
+* Simplified watchtowers/accelerators: No requirement to equip watchtowers with privileged key material, and no value to steal by those watchtowers
+* Multiparty smart contracts: LN BOLTs can be revamped to remove value-sapping anchors.
+* LN-Symmetry: During the contest period, no anchor value can be sourced endogenously.
+* [https://bitcoin.stackexchange.com/questions/100537/what-is-spacechain Spacechains]
+* Ark transactions, for anchors and connector outputs
+* Timeout Trees, for anchors and connector outputs
+
+===Related Work===
+
+[https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-January/021334.html SIGHASH_GROUP] style proposals are an alternative method of bringing funds to a transaction without involving CPFP by enacting a softfork. Making these pin-resistant may require follow-on policy work, or [https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-May/020458.html more general covenants] to directly stop pins we want to avoid. The drawback of these are the necessity of a softfork.
+
+[https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-September/018168.html Transaction Sponsors] is a softfork proposal to allow transactions to
+sponsor transactions with no explicit relationship in the classical UTXO
+model. Ephemeral Dust with an un-keyed anchor can be viewed as a type of opt-in transaction sponsors implemented
+purely in policy.
+
+Using a 0-value CPFP anchors is not a new idea, see:
+
+* [https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-May/015931.html LN-dev discussion on 0-value anchors]
+
+The discussion lacked a solution to the issue of the dust entering into the UTXO set
+causing negative externalities.
+
+==Definitions==
+
+Ephemeral dust: An output with dust value which is immediately spent by a child transaction.
+
+Ephemeral dust transaction: A transaction that has an ephemeral dust output.
+
+==Specification==
+
+When received by a peer for inclusion to the mempool an ephemeral dust transaction MUST:
+
+* Be an otherwise valid transaction
+* Be only considered for mining with ephemeral dust spent
+* Have only one ephemeral dust output (output values which would normally cause rejection)
+
+or will be rejected by policy. All other policy checks are left in place.
+These constraints apply only to mempool policy. Blocks are not
+invalidated by breaking these policy-only rules as they have no bearing on consensus.
```

Comment:
```suggestion

As an exception to the dust rule, a transaction with dust outputs will be considered for acceptance to a node's mempool if:

* It has arrived as part of a package of two or more transactions
* Other transactions in that package spend every dust output
* The transaction(s) spending the dust outputs have a higher effective fee-rate than the transaction creating the dust

This specification is aspirational. Actual implementations may place further restrictions for implementation reasons.
```

Rational: I think it's more clear and less wordy if we explain the specification in terms of how ephemeral dust is a *relaxation* of the dust rule, allowing transactions to do something they otherwise could not have done.

Reply:
Took the clarity suggestion on making it a more straight forward relaxation.

I also rewrote the section hoping to stay as close as possible to the ideal, which is "don't make a block template that would result in dust entering UTXO set"